akka-stream-tests/src/test/scala/akka/stream/scaladsl/SetupSpec.scala

```diff
@@ -207,7 +207,7 @@ class SetupSpec extends StreamSpec {
         }
         .named("my-name")
 
-      Source.empty.runWith(sink).flatMap(identity).flatMap(identity).futureValue shouldBe Some("setup-my-name")
+      Source.empty.runWith(sink).flatMap(identity).flatMap(identity).futureValue shouldBe Some("setup-setup-my-name")
```

Comment:
It's odd that we are using `flatMap(identity)` everywhere here. It will be better to use `flatten`. 

Anyway, I can PR to change that once we merge this one. Unless you want to tackle that in this PR. Up to you @armanbilge.

Reply:
@octonato I feel that there is a reason, but I can't quite remember. I guess you'll run into it if you try to replace ;)

Reply:
I did and it worked as expected. I did learn something though, `flatten` doesn't require an execution context. I've never noticed it. 

In any case, if there is a reason for that, it will be better to document it.

Reply:
IIRC `flatten` is essentially `flatMap(identity)(ExecutionContext.parisitic)`

Reply:
@armanbilge, it uses the `InternalCallbackExecutor` which does basically the same, but it's internal to `scala.concurrent`

@johanandren, I don't think that there is any reason for using `flatMap(identity)`. This is done on the Futures returned by the `runWith`. That has nothing to do with any of the code involved in the test.

Reply:
Agreed, nothing to do with the test.